mlrun/serving/states.py

```diff
@@ -3466,6 +3481,92 @@ def get_monitored_steps(self) -> dict[str, "MonitoredStep"]:
         }
 
 
+class HubTaskStep(TaskStep):
+    """hub task execution step, runs a class or handler from a hub"""
+
+    kind = "hub_task"
+    _dict_fields = TaskStep._dict_fields + ["hub_step_class_name", "requirements"]
+
+    def __init__(
+        self,
+        class_name: Optional[Union[str, type]] = None,
+        hub_step_class_name: Optional[str] = None,
+        class_args: Optional[dict] = None,
+        handler: Optional[str] = None,
+        name: Optional[str] = None,
+        after: Optional[list] = None,
+        full_event: Optional[bool] = None,
+        function: Optional[str] = None,
+        responder: Optional[bool] = None,
+        input_path: Optional[str] = None,
+        result_path: Optional[str] = None,
+        model_endpoint_creation_strategy: Optional[
+            schemas.ModelEndpointCreationStrategy
+        ] = schemas.ModelEndpointCreationStrategy.SKIP,
+        endpoint_type: Optional[schemas.EndpointType] = schemas.EndpointType.NODE_EP,
+        requirements: Optional[list] = None,
+    ):
+        super().__init__(
+            class_name=class_name,
+            class_args=class_args,
+            handler=handler,
+            name=name,
+            after=after,
+            full_event=full_event,
+            function=function,
+            responder=responder,
+            input_path=input_path,
+            result_path=result_path,
+            model_endpoint_creation_strategy=model_endpoint_creation_strategy,
+            endpoint_type=endpoint_type,
+        )
+        self.hub_step_class_name = hub_step_class_name
+        self.requirements = requirements
+
+    @staticmethod
+    @contextmanager
+    def hub_step_tempdir():
+        """
+        Create a temporary directory named 'hub_step' inside the system temp directory.
+        Directory is cleaned up automatically when the context exits.
+        """
+        base = tempfile.gettempdir()
+        path = os.path.join(base, "hub_steps")
+        if os.path.exists(path):
+            shutil.rmtree(path)
+        os.makedirs(path, exist_ok=True)
+        try:
+            yield path
+        finally:
+            shutil.rmtree(path, ignore_errors=True)
+
+    def init_object(self, context, namespace, mode="sync", reset=False, **extra_kwargs):
+        self.context = context
+        self._async_object = None
+        if not self._is_local_function(context):
+            return
+
+        with self.hub_step_tempdir() as local_path:  # self-cleaning tmp dir util
+            hub_step = mlrun.get_hub_step(self.class_name, local_path=local_path)
+            mod = hub_step.module()
+
+        if self.handler and not self.hub_step_class_name:
```

Comment:
Is HubStep can be an handler and not a class?
And if it does lets resolve it also from the metadata of the hub step API.